### DISCUSSION TITLE
Failing badge status for passing runs and filters not working correctly

### DISCUSSION THREAD

**Discussion by OKaluza** [community/community#24986] [Actions]
<p>Hi, anyone know why this status badge is showing “Failing”?<br>
<a href="https://github.com/lavavu/LavaVu/actions?query=workflow%3ADeploy" rel="noopener nofollow ugc"><img src="https://github.com/lavavu/LavaVu/workflows/Deploy/badge.svg" width="108" height="20"></a></p>
<p>When the associated workflow is passing: <a href="https://github.com/lavavu/LavaVu/actions?query=workflow%3ADeploy" rel="noopener nofollow ugc">here</a></p>
<p>Notice passing status of most recent actions.</p>
<p>Another issue I notice: runs are also appearing incorrectly at the above linked workflow from different workflows, set filter is “Deploy” but is also showing runs from the “OSMesa” workflow.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/okaluza">@OKaluza</a>,</p>
<p>There is an earlier reported topic about the similar issue:</p><aside class="quote" data-post="1" data-topic="139825">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/bug-workflow-badge-always-shows-failure-if-workflow-name-has-dashes-maybe/139825">[BUG] Workflow badge always shows failure if workflow name has dashes maybe?</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-actions/41"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #2188ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Actions—automation for all of your development workflows, now with world-class CI/CD.">GitHub Actions</span></a>
  </div>
  <blockquote>
    Hi folks, 
I had a workflow that was always reporting failure state even though it was succeeding. The workflow name AND file name were both the same: build-release-containers. The file had the yml extension, though. 
Once I changed the workflow name to something more natural, like “Build Releae Containers”, the badge began to properly show the success state. 
Maybe a bug?
  </blockquote>
</aside>

<p>Sometimes, we also encounter the same problem. After waiting a moment and refresh (or reopen) the web page, the status badge generally can correctly show the latest status.</p>

**Comment by OKaluza** [comment]
<p>Thanks for the response <a class="mention" href="/brightran">@brightran</a>. I did give it a while and a few refreshes, but I think it is a deeper problem because the badge is still showing “Failing”, and it’s a couple of days since the action passed now.</p>

**Comment by nethgato** [comment, marked as answer]
<p>Hey there <a class="mention" href="/okaluza">@OKaluza</a> :wave:</p>
<p>Pinged our team on this one and discovered that the latest run for the  <code>Deploy</code>  workflow  <strong>on the default branch</strong>  was cancelled, which is why the badge is showing as failed.</p>
<p>However, as <a class="mention" href="/brightran">@brightran</a> mentioned, there are some other threads that show evidence of similar behavior under different circumstances. While this particular thread has an answer, I want to make mention of some efforts going on internally, here.</p>
<p>There is refresh work happening that should help with erroneous <code>failing</code> tags being added to successful runs. There is no ETA that I can provide on this effort, and it specifically wouldn’t solve the problem here, but it should make that inadvertent tagging no longer a problem, once completed.</p>
<p>Thanks so much for raising this! :bow:</p>

**Comment by michhyun1** [comment]
<p>I am seeing the same issue with my badge:</p><aside class="onebox githubfolder">
  <header class="source">
      <a href="https://github.com/aws/amazon-chime-sdk-js/workflows/CI%20Workflow/badge.svg" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://github.com/aws/amazon-chime-sdk-js/workflows/CI%20Workflow/badge.svg" target="_blank" rel="noopener nofollow ugc"></a></h3>

<p><a href="https://github.com/aws/amazon-chime-sdk-js/workflows/CI%20Workflow/badge.svg" target="_blank" rel="noopener nofollow ugc">//github.com/aws/amazon-chime-sdk-js/workflows/CI%20Workflow/badge.svg</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>I wanted to know why it shows failing, even though the last run of the CI Workflow is passing. <a href="https://github.com/aws/amazon-chime-sdk-js/actions?query=workflow%3A%22CI+Workflow%22" class="inline-onebox" rel="noopener nofollow ugc">Actions · aws/amazon-chime-sdk-js · GitHub</a></p>

**Comment by OKaluza** [comment]
<p>What I didn’t understand was the badge shows the status of the master branch only, my workflow was running on release in tagged branches, but the badge was displaying the status of the action when run on the master branch (which was some old failed test run).<br>
I fixed it by specifying the branch with ?branch=releasetag in the badge url, of course unfortunately the badge url needs to be updated with every release.</p>
<p>It looks like the same thing with your repo, if you look at the master branch the status is failing, so the badge is correct: <a href="https://github.com/aws/amazon-chime-sdk-js/actions?query=workflow%3A%22CI+Workflow%22+branch%3Amaster" class="inline-onebox" rel="noopener nofollow ugc">Actions · aws/amazon-chime-sdk-js · GitHub</a> if you specify the name of the most recent passing branch from a pull request it shows passing: <a href="https://github.com/aws/amazon-chime-sdk-js/workflows/CI%20Workflow/badge.svg?branch=hotfix-script" rel="noopener nofollow ugc">https://github.com/aws/amazon-chime-sdk-js/workflows/CI%20Workflow/badge.svg?branch=hotfix-script</a></p>
<p>It would actually be really useful to have a badge that reported the status of the last run of a particular action, regardless of branch, or a way to specify a wildcard branch.</p>

**Comment by michhyun1** [comment]
<p>So it defaults to using master branch, that I understand. Then why can’t I use this option? <a href="https://docs.github.com/en/actions/managing-workflow-runs/adding-a-workflow-status-badge#using-the-event-parameter" class="inline-onebox" rel="noopener nofollow ugc">Adding a workflow status badge - GitHub Docs</a></p>
<p>For my understanding, my workflow is triggered from a ‘pull_request’, but how come if I do <a href="https://github.com/aws/amazon-chime-sdk-js/workflows/CI%20Workflow/badge.svg?event=pull_request" rel="noopener nofollow ugc">https://github.com/aws/amazon-chime-sdk-js/workflows/CI%20Workflow/badge.svg?event=pull_request</a>, it doesnt show the status of the last time this workflow was triggered by this event, regardless of the branch?</p>

**Comment by OKaluza** [comment]
<aside class="onebox githubfolder">
  <header class="source">
      <a href="https://github.com/aws/amazon-chime-sdk-js/workflows/CI%20Workflow/badge.svg?event=pull_request&amp;branch=hotfix-script" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://github.com/aws/amazon-chime-sdk-js/workflows/CI%20Workflow/badge.svg?event=pull_request&amp;branch=hotfix-script" target="_blank" rel="noopener nofollow ugc"></a></h3>

<p><a href="https://github.com/aws/amazon-chime-sdk-js/workflows/CI%20Workflow/badge.svg?event=pull_request&amp;branch=hotfix-script" target="_blank" rel="noopener nofollow ugc">//github.com/aws/amazon-chime-sdk-js/workflows/CI%20Workflow/badge.svg</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
<p>
It seems when using the event filter, it still insists on a specific branch, defaulting to master. Because the pull request happens in a branch, it doesn’t show unless you explicitly add the branch filter.</p>
<p>Again, I think it makes sense to be able to specify “any” or “*” as a branch, or even better and more intuitive: default to no branch filter at all and only behave the way it does now when a branch parameter is used. <a class="mention" href="/nethgato">@nethgato</a> any chance of this making it to the devs? Is it worth posting it somewhere else as a feature request?</p>

**Comment by michhyun1** [comment]
<p>Having the Branch filter be able to use “*” would fit this use case. I’m not sure why it isnt included already, as having a CI workflow means that the CI workflow would run on the remote branch that submits the PR. The name of the remote branch created for the purpose of the PR cannot be known beforehand, so how could I use the remote branch as part of this query parameter?</p>
<p>There should be a way to show the status of ALL RUNS of an arbitrary Github Actions workflow, not just the runs that occur in a specified branch, right? <a class="mention" href="/nethgato">@nethgato</a></p>

**Comment by beeb** [comment]
This is still an issue today @nethgato 

https://github.com/beeb/remote-wol-rs/actions/workflows/build.yml

![2023-01-23_06-37](https://user-images.githubusercontent.com/703631/213972747-946004f8-90a8-42d3-9210-5407b932fc01.png)

**Reply by FrolovVyacheslav** [reply]
+1

**Comment by drewtadams** [comment]
still seeing this

![Screenshot 2025-06-05 at 7 38 24 AM](https://github.com/user-attachments/assets/bb9932d4-62b6-4411-a32d-fd6e58653b59)

**Comment by geraldolsribeiro** [comment]
<img width="556" height="590" alt="image" src="https://github.com/user-attachments/assets/06147fcc-a3db-4e54-8f41-3ccb7fb23e04" />

Still seeing "failling" besides the green in the workflow and the green tick on commit
